### PR TITLE
upgrading to react-leaflet v3 and corresponding changes etc

### PR DESCRIPTION
**Pull request by moontrip**
This is an attempted to adopt react-leaflet v3.

- TL:DR
From my tests, this works for all map related stories and functionality implemented at the map. Revising codes to work with this upgrades. Tidying some codes, especially Donut/Chart makers.

- To-do
a) leaflet upgrade is not applied yet. Will do that.
b) Some functionality regarding ref, e.g., flyTo and thumbnail could not be tested as it should be done through web-eda's mapViz: I have tried to make it happen but untested. Thus, it would be done with web-eda.

- Installation note
There are many changes in packages so it is recommended to remove ./node_modules directory and yarn install for this work, just in case.

- some details for compiling issue
a) add/upgrade packages:
`react-leaflet` v2 -> v3
`@types/leaflet`
`react-leaflet-drift-marker`: replacing `leaflet-drift-marker`
`@babel/plugin-proposal-nullish-coalescing-operator`

b) changes for successful compiling, including storybook
i) nullish coalescing operator issue from react-leaflet v3 and CRA
browserslist at package.json:
from
`
  "browserslist": [
    "extends @veupathdb/browserslist-config"
  ],
`
to
`
  "browserslist": [
    ">0.2%",
    "not dead",
    "not op_mini all"
  ],

`
ii) changed `.storybook/main.js` for running storybook correctly

### CODE REVIEW

**Comment by moontrip** [comment]
Tried to upgrade leaflet to the latest version, 1.7.1. Initially I thought there were some conflicts with current map implementations but after testing extensively it was found to be just my mistakes. Seems to be fine with this new version overall.

**Review by bobular — COMMENTED** [review]
Had a look at the code. A lot of work here! Looks good, but I would like to do some testing (storybook and EDA) and can't right now, sorry!

**Comment by moontrip** [comment]
@bobular Thank you for your interest 👍 I think that all map-related stories seem to work fine within this web-components, however, I could not test web-eda part because I could not resolve the issue of allowing `react-leaflet-v3` at web-eda: keep showing v2 in there with `yarn link`. Didn't try `npm-pack-here`.

**Comment by bobular** [comment]
Oh, I see I haven't approved yet! Is it time to come out of `Draft` status?

**Comment by bobular** [comment]
Oh, I remember now, https://github.com/VEuPathDB/web-components/pull/347 is the active PR.
Please could you close this, if appropriate?

**Comment by moontrip** [comment]
@bobular yes indeed. #347 is the latest PR, which was originally branched out from this PR. Thus, this PR is no longer valid. I will close this.
